velox/dwio/dwrf/common/FileMetadata.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+
+#include "velox/dwio/common/Common.h"
+#include "velox/dwio/dwrf/common/Common.h"
+#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
+#include "velox/dwio/dwrf/common/wrap/orc-proto-wrapper.h"
+
+namespace facebook::velox::dwrf {
+
+enum DwrfFormat : uint8_t { kDwrf, kOrc };
+
+class ProtoWrapperBase {
+ protected:
+  DwrfFormat format_;
+  void* impl_;
+
+ public:
+  ProtoWrapperBase() = delete;
+
+  ProtoWrapperBase(DwrfFormat format, void* impl)
+      : format_{format}, impl_{impl} {}
+
+  DwrfFormat format() const {
+    return format_;
+  }
+
+  inline void* rawProtoPtr() const {
+    return impl_;
+  }
+};
+
+class PostScript {
```

Comment:
It worths also using a wrapper with this because of

1. The set bit information in proto object
2. Size of the object is so large that it might not me optimal to return it directly